src/lib/detection_algorithms/cntk/exporter.js

```diff
@@ -0,0 +1,158 @@
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  `${frameFileName}.bboxes.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, `${frameFileName}.bboxes.tsv`)),
+                function determineWritePath(cb){
+                    if( !tags.length){ 
+                        cb(null, self.negDirPath);
+                    } else {
+                         if (self.testFrameIndecies.includes(self.posFrameIndex)){
+                             self.posFrameIndex++;
+                             cb(null, self.testDirPath);
+                         } else {
+                             self.posFrameIndex++;
+                             cb(null, self.posDirPath);
+                         }
+                    }
+                },
+                function saveImage(exportPath, cb) {
+                    var imageFilePath  = path.join(exportPath, frameFileName);
+                    fs.writeFileSync(imageFilePath, frameBuffer);
+                    cb(null, exportPath); 
+                },
+                function saveBboxes(exportPath, cb) {
+                    if (exportPath !== self.negDirPath){
+                        var labelData = '';
+                        var bboxesData = '';
+                        for (var i in tags) {
+                            var tag = tags[i];
+                            labelData  += `${tag.class}\n`;
+                            bboxesData += `${tag.x1}\t${tag.y1}\t${tag.x2}\t${tag.y2}\n`;
+                        }
+                        var labelFilePath  = path.join(exportPath, `${frameFileName}.bboxes.labels.tsv`);
+                        var bboxesFilePath  = path.join(exportPath, `${frameFileName}.bboxes.tsv`);
+                        fs.writeFileSync(labelFilePath, labelData);
+                        fs.writeFileSync(bboxesFilePath, bboxesData);
+                    } 
+                    cb();
+                }
+            ],
+            function(err) {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+                resolve();
+            });
+        });
+    }       
+
+    //random set http://stackoverflow.com/questions/2380019/generate-unique-random-numbers-between-1-and-100 there has to be a set way of doing this
+    function generateTestIndecies(percent, cb) {
+       var testIndecies = [];
+       while(testIndecies.length < Math.ceil(percent * self.posFramesCount)){
+            var randomnumber = Math.ceil(Math.random() * self.posFramesCount)
+            if(testIndecies.indexOf(randomnumber) > -1) continue;
+            testIndecies[testIndecies.length] = randomnumber;
+        }
+        cb(null, testIndecies);
+    } 
+    // http://stackoverflow.com/questions/21194934/node-how-to-create-a-directory-if-doesnt-exist
+    function ensureDirExists(path, cb) {
```

Comment:
Move it to a new utils module..
You can place it under lib/detection_algorithms
And then do require('../detectionUtil.js') both from Yolo and from here.